reproducibility_project/src/analysis/equlibration.py

```diff
@@ -76,10 +80,54 @@ def trim_non_equilibrated(
         discarding more data.
 
     """
-    [truth, t0, g] = is_equilibrated(a_t, threshold=threshold, nskip=nskip)
+    [truth, t0, g, Neff] = is_equilibrated(
+        a_t, threshold=threshold, nskip=nskip
+    )
     if not truth:
         raise ValueError(
             f"Data with a threshold of {threshold} is not equilibrated!"
         )
 
-    return [a_t[t0:], g, t0]
+    return [a_t[t0:], t0, g, Neff]
+
+
+def plot_data_with_t0_line(
+    filename: str,
+    a_t: npt.ArrayLike,
+    threshold: float = 0.0,
+    overwrite: bool = False,
+    plt_kwargs: dict = None,
+) -> None:
+    """Plot data with a vertical line at beginning of equilibration.
+
+    Parameters
+    ----------
+    a_t : numpy.typing.ArrayLike
+        1-D time dependent data
+    threshold : float, optional, default=0.0
+        Threshold to error out on if threshold fraction of data is not equilibrated.
+    overwrite : bool, optional, default=False
+        Do not write to filename if a file already exists with the same name.
+        Set to True to overwrite exisiting files.
+    plt_kwargs : dict, optional, default=None
+        keyword dictionary for matplotlib.pyplot.plot command.
+    """
+    scale = 1.25
+    path = pathlib.Path(filename)
+    if path.is_file() and not overwrite:
+        raise FileExistsError(
+            f"Cannot write {path.name}, file already exists at: {path.absolute()}. Set overwrite=True to replace file."
+        )
+
+    _, t0, g, Neff = is_equilibrated(a_t, threshold=threshold, nskip=1)
+
+    ymin = np.min(a_t) * scale
+    ymax = np.max(a_t) * scale
+
+    plt.plot(
+        a_t,
+    )
+    plt.vlines(x=t0, ymin=ymin, ymax=ymax)
```

Comment:
It might be better to do a dashed line style and a color other than the blue matplot lib defaults to.

Reply:
Agreed. I think we can hardcode the dash, but if we want the color to change with the rc file (#74) we can use [cycler](https://matplotlib.org/stable/tutorials/intermediate/color_cycle.html)

Reply:
I like this idea. Ill begin with a basic hardcoded one, but we can make this a bit more programmatic later.